README.md

```diff
@@ -198,6 +198,15 @@ A `TraceApi` object is guaranteed to be returned by both of these calls, even if
 
 A fully detailed overview of the `TraceApi` object is available [here](doc/trace-api.md).
 
+## How does automatic tracing work?
+
+Node.js provides concurrency through as asynchronous execution model. To be able to keep track of the current request across asynchronous execution boundaries we use the [async-listener][] module. It modifies the core async APIs in Node.js, and Promises, to save and restore the 'execution context' across the async bounaries. In addition, we instrument frameworks (like express, hapi, etc. if loaded) to create request contexts for the execution of the incoming requests. Similarly, we instrument RPC libraries (like mysql, redis, outbound http, etc.) to keep track of the latency of these operations and to correlate them to the incoming request.
```

Comment:
A few fixes/proposals:
- First sentence: as -> an
- Last two sentences should indicate that we utilize the saved/restored execution context to instrument frameworks and RPC libraries.
- Nit: Last sentence should also end with "incoming requests" for the sake of uniformity